### PR TITLE
feat(spans): Add event ID to extracted spans

### PR DESCRIPTION
**Pull request by jjbayer**
True standalone spans will not have an associated event ID. But because the current spans dataset expects it, add it to the kafka message of spans extracted from transactions for now.

#skip-changelog

### CODE REVIEW

**Review by olksdr — APPROVED** [review]
lgtm!

Quick question: will consumers process these message with new field correctly?

File: relay-server/src/actors/store.rs [hunk 1]
```diff
@@ -1032,12 +1035,17 @@ struct CheckInKafkaMessage {
...
     span: serde_json::Value,
     /// Time at which the span was received by Relay.
     start_time: u64,
     /// The project id for the current span.
     project_id: ProjectId,
+    /// The event id for the current span.
+    ///
+    /// Once spans are truly standalone, this field will be omitted.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_id: Option<EventId>,
```
Comment (iker-barriocanal) [inline relay-server/src/actors/store.rs:1048]: Q: is this an option to make the transition to no event_id smoother?

**Review by iker-barriocanal — APPROVED** [review]

**Comment by jjbayer** [comment]
> lgtm!
> 
> Quick question: will consumers process these message with new field correctly?

@olksdr yep AFAIK the [consumer](https://github.com/getsentry/sentry/blob/4349a91fcac183474020f850a241d5ef22ce52f4/src/sentry/spans/consumers/process/factory.py#L36) ignores fields it does not know, but I'll let @phacops confirm.

File: relay-server/src/actors/store.rs (hunk 1 above)
Comment (jjbayer) [reply relay-server/src/actors/store.rs:1048]: That was the idea, yes. `handle_store_envelope` produces kafka messages for span items in an envelope regardless of whether the envelope contains an event.

**Review by phacops — APPROVED** [review]

**Comment by phacops** [comment]
Indeed, we sort of ignore the field right now because we're not even looking at the right place for it.

https://github.com/getsentry/sentry/blob/8e90edebdd5775824ae4880f5e5fdd28a012036a/src/sentry/spans/consumers/process/factory.py#L48

**Pull request merged**